converter/internal/staticconvert/staticconvert.go

```diff
@@ -88,9 +90,54 @@ func AppendStaticPrometheus(f *builder.File, staticConfig *config.Config) diag.D
 		//   metrics config name = "agent"
 		//   scrape config job_name = "test_prometheus"
 		//
-		//   results in two prometheus.scrape components with the label "agent_test_prometheus"
-		newDiags := prometheusconvert.AppendAll(f, promConfig, instance.Name)
-		diags = append(diags, newDiags...)
+		//   results in two prometheus.scrape components with the label "metrics_agent_test_prometheus"
+		diags.AddAll(prometheusconvert.AppendAll(f, promConfig, "metrics_"+instance.Name))
+	}
+
+	return diags
+}
+
+func appendStaticPromtail(f *builder.File, staticConfig *config.Config) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	if staticConfig.Logs == nil {
+		return diags
+	}
+
+	for _, logConfig := range staticConfig.Logs.Configs {
+		promtailConfig := &promtail_config.Config{
+			Global:          promtail_config.GlobalConfig{FileWatch: staticConfig.Logs.Global.FileWatch},
+			ServerConfig:    server.Config{Disable: true},
+			ClientConfig:    client.Config{},
+			ClientConfigs:   logConfig.ClientConfigs,
+			PositionsConfig: logConfig.PositionsConfig,
+			ScrapeConfig:    logConfig.ScrapeConfig,
+			TargetConfig:    logConfig.TargetConfig,
+			LimitsConfig:    logConfig.LimitsConfig,
+			Options:         promtail_config.Options{},
+			Tracing:         tracing.Config{Enabled: false},
+			WAL:             wal.Config{Enabled: false},
+		}
+
+		// We need to set this when empty so the promtail converter doesn't think it has been overridden
+		if promtailConfig.Global == (promtail_config.GlobalConfig{}) {
```

Comment:
```suggestion
		if promtailConfig.Global == promtail_config.GlobalConfig{} {
```
Do you need the parens?

Reply:
you do need it to do this inline here or you receive

`expected ';', found '{'`